runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -40,7 +42,7 @@
  *  TODO: Generalize to all SMR objects.
  */
 @Slf4j
-public class CheckpointWriter<T extends Map> {
+public class CheckpointWriter<T extends StreamingMap> {
```

Comment:
StreamingMap is too low level to expose on a public API. I think `ICorfuSMR` is more appropriate, `ISMRMap`.

Reply:
Consumed via `ISMRMap` now.

Reply:
Turns out we cannot use `ISMRMap`, since this interface is only consumed only `SMRMap` and not `CorfuTable`. Making `CorfuTable` inherit from `ISMRMap` makes little sense as it contains bunch of logic only specific to `SMRMap`.

If we were to remove `SMRMap`, this change ought to be possible. For now I ensured that `StreamingMap` does not expose any Corfu specific internal -- it just contains `Stream<Map.Entry<K, V>> entryStream()`.